### PR TITLE
RDoc-1244 Document that calling 'put' will not copy attachments in patching

### PR DESCRIPTION
**Pull request by haludi**
* [Document issue: RDoc-1244](http://issues.hibernatingrhinos.com/issue/RDoc-1244)
* [Origin issue: RavenDB-11391](http://issues.hibernatingrhinos.com/issue/RavenDB-11391)

### CODE REVIEW

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 1]
```diff
@@ -210,6 +210,24 @@ Renaming property supported only by the non-typed APIs
...
+
+Adding a new document supported only by the non-typed APIs
+{CODE-TABS}
+{CODE-TAB:csharp:Session-syntax-untyped add_document_session@ClientApi\Operations\Patches\PatchRequests.cs /}
+{CODE-TAB:csharp:Operations-syntax add_document_store@ClientApi\Operations\Patches\PatchRequests.cs /}
+{CODE-TABS/}
+
+###Clone document
+
+Coloning document supported only by the non-typed APIs
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: `Cloning`

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 2]
```diff
@@ -210,6 +210,24 @@ Renaming property supported only by the non-typed APIs
 {CODE-TAB:csharp:Operations-syntax rename_property_age_store@ClientApi\Operations\Patches\PatchRequests.cs /}
 {CODE-TABS/}
 
+###Add document
+
+Adding a new document supported only by the non-typed APIs
```
Comment (ayende) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: document *is* supported

Comment (ayende) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: untyped

Comment (haludi) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: So is it make sense to change all the equivalent text to keep consistency?

File: Documentation/4.0/Raven.Documentation.Pages/server/kb/javascript-engine.markdown [hunk 3]
```diff
@@ -48,7 +48,7 @@ RavenDB introduced a set of predefined function, in addition to Jint's ECMAScrip
...
 | **del(documentId)** | `void` | Used in patches, deletes the document with the given ID. |
-| **put(documentId, document, [optional]changeVectorString)** | `Task` | Used in patches, saves a document with the given ID. This function can also be used for updating documents. In order to create a new document, it's possible to use "[collectionPrefix]/" [Server-Side ID](../../server/kb/document-identifier-generation#server-side-id) notation in order to generate a new document ID |
+| **put(documentId, document, [optional]changeVectorString)** | `Task` | Used in patches, create or update a document with the given ID. In order to generate a new document ID it's possible to use "[collectionPrefix]/" [Server-Side ID](../../server/kb/document-identifier-generation#server-side-id) notation<sup>[[ex]](../../client-api/operations/patching/single-document#add-document)</sup>. <br/>This function can also be used to clone an existing document (Note: Attachments will not be attached to the clone)<sup>[[ex]](../../client-api/operations/patching/single-document#clone-document)</sup>.  |
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/server/kb/javascript-engine.markdown]: create or update => creates or updates

[the remark this answers is not in the document]
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/server/kb/javascript-engine.markdown]: Looks nice IMO. @arekpalinski what do you think?

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 4]
```diff
@@ -725,6 +725,46 @@ where id() in ($ids)
 
                 #endregion
             }
+
+            using (var store = new DocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    #region add_document_session
+                    session.Advanced.Defer(new PatchCommandData("dogOwners/A-1", null,
```
Comment (ppekrol) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: If possible, we are trying to stick to Northwind examples. Can we do that in this case?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown [hunk 5]
```diff
@@ -210,6 +210,24 @@ Renaming property supported only by the non-typed APIs
...
+
+###Clone document
+
+Coloning document supported only by the non-typed APIs
+{CODE-TABS}
+{CODE-TAB:csharp:Session-syntax-untyped clone_document_session@ClientApi\Operations\Patches\PatchRequests.cs /}
+{CODE-TAB:csharp:Operations-syntax clone_document_store@ClientApi\Operations\Patches\PatchRequests.cs /}
+{CODE-TABS/}
+
+* Cloning document with attachments will generate a new ducoment without attachments
```
Comment (ppekrol) [inline Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: ducoment -> document

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 1 above)
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: Maybe: `Document cloning is supported only by a untyped APIs` ?

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 5 above)
Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: Let's put it in nice frame.

Use:
```
{INFO:Cloning & Attachments} 
Source document attachments will not be cloned during the cloning operation.
{/INFO}
```

Comment (ppekrol) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: User can copy them after the document was cloned? Or I've mixed sth?

If this is true, then I suggest adding an example inside the {INFO} block

**Comment by ppekrol** [comment]
IIRC this article was copied to 4.1 docs already (@aviv86 ?) We need to remember to apply the changes there also.

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 1 above)
Comment (haludi) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: It talks about patching here and maybe I am wrong but the type APIs let you to

* Increment field
* Set field
* Manipulate field array

So cloning document in patch is possible only throw the untyped patch APIs
Is it correct?
Maybe I should give more explanation

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 5 above)
Comment (haludi) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: If I remember right @ayende talked about adding this feature later

Comment (haludi) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: And I am afraid that
> "Source document attachments will not be cloned during the cloning operation." 

can be understood as the clone document will be attached to the old attachment (the attachment didn't clone) so what do you say if I will write it like this 
>"Source document attachments will not be cloned <b>or attached</b> during the cloning operation."

File: Documentation/4.0/Raven.Documentation.Pages/server/kb/javascript-engine.markdown (hunk 3 above)
[the remark this answers is not in the document]
Comment (arekpalinski) [reply Documentation/4.0/Raven.Documentation.Pages/server/kb/javascript-engine.markdown]: +1

**Comment by haludi** [comment]
@ppekrol : So I will change all 
>non-typed APIs 

to 
>untyped APIs 

including the titles, O.k.?

image

**Comment by ppekrol** [comment]
Let's stay with non-typed then

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 1 above)
Comment (haludi) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: done___

File: Documentation/4.0/Raven.Documentation.Pages/server/kb/javascript-engine.markdown (hunk 3 above)
Comment (haludi) [reply Documentation/4.0/Raven.Documentation.Pages/server/kb/javascript-engine.markdown]: done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 4 above)
Comment (haludi) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown (hunk 5 above)
Comment (haludi) [reply Documentation/4.0/Raven.Documentation.Pages/client-api/operations/patching/single-document.dotnet.markdown]: done

**Comment by ppekrol** [comment]
Can you resolve conflicts?

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 6]
```diff
@@ -725,6 +725,46 @@ where id() in ($ids)
 
                 #endregion
             }
+
+            using (var store = new DocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    #region add_document_session
+                    session.Advanced.Defer(new PatchCommandData("employees/A-1", null,
```
Comment (ppekrol) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 7]
```diff
@@ -725,6 +725,46 @@ where id() in ($ids)
...
+                        new PatchRequest
+                        {
+                            Script = "put('orders/', { Employee: id(this) });",
+                        }, null));
+
+                    session.SaveChanges();
+                    #endregion
+
+                    #region clone_document_session
+                    session.Advanced.Defer(new PatchCommandData("employees/A-1", null,
```
Comment (ppekrol) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 8]
```diff
@@ -725,6 +725,46 @@ where id() in ($ids)
...
+                        {
+                            Script = "put('employees/', this);",
+                        }, null));
+
+                    session.SaveChanges();
+                    #endregion
+                }
+
+                #region add_document_store
+                store.Operations.Send(new PatchOperation("employees/A-1", null, new PatchRequest
```
Comment (ppekrol) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 9]
```diff
@@ -725,6 +725,46 @@ where id() in ($ids)
...
+                {
+                    Script = "put('orders/', { Employee: id(this) });",
+                }));
+                #endregion
+
+                #region clone_document_store
+                store.Operations.Send(new PatchOperation("employees/A-1", null, new PatchRequest
```
Comment (ppekrol) [inline Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 10]
```diff
@@ -734,6 +734,46 @@ where id() in ($ids)
...
+
+                #region add_document_store
+                store.Operations.Send(new PatchOperation("employees/A-1", null, new PatchRequest
+                {
+                    Script = "put('orders/', { Employee: id(this) });",
+                }));
+                #endregion
+
+                #region clone_document_store
+                store.Operations.Send(new PatchOperation("employees/A-1", null, new PatchRequest
```
Comment (ppekrol) [inline Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 11]
```diff
@@ -734,6 +734,46 @@ where id() in ($ids)
...
+                        {
+                            Script = "put('employees/', this);",
+                        }, null));
+
+                    session.SaveChanges();
+                    #endregion
+                }
+
+                #region add_document_store
+                store.Operations.Send(new PatchOperation("employees/A-1", null, new PatchRequest
```
Comment (ppekrol) [inline Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 12]
```diff
@@ -734,6 +734,46 @@ where id() in ($ids)
...
+                        new PatchRequest
+                        {
+                            Script = "put('orders/', { Employee: id(this) });",
+                        }, null));
+
+                    session.SaveChanges();
+                    #endregion
+
+                    #region clone_document_session
+                    session.Advanced.Defer(new PatchCommandData("employees/A-1", null,
```
Comment (ppekrol) [inline Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs [hunk 13]
```diff
@@ -734,6 +734,46 @@ where id() in ($ids)
 
                 #endregion
             }
+
+            using (var store = new DocumentStore())
+            {
+                using (var session = store.OpenSession())
+                {
+                    #region add_document_session
+                    session.Advanced.Defer(new PatchCommandData("employees/A-1", null,
```
Comment (ppekrol) [inline Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: should be: `employees/1-A`

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 6 above)
Comment (haludi) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 7 above)
Comment (haludi) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 8 above)
Comment (haludi) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 9 above)
Comment (haludi) [reply Documentation/4.0/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 10 above)
Comment (haludi) [reply Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 11 above)
Comment (haludi) [reply Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 12 above)
Comment (haludi) [reply Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

File: Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs (hunk 13 above)
Comment (haludi) [reply Documentation/4.1/Samples/csharp/Raven.Documentation.Samples/ClientApi/Operations/Patches/PatchRequests.cs]: done

**Pull request merged**